src/icp_canister_backend/tests/rewards_tests.rs

```diff
@@ -90,73 +87,431 @@ fn test_reward_rate_increase_decrease_during_episodes() {
+    let expected_rewards_b = reward_amount.clone() / Nat::from(3u64); 
+    
+    let tolerance_a = expected_rewards_a.clone() / Nat::from(10000u64);
+    let tolerance_b = expected_rewards_b.clone() / Nat::from(10000u64);
+    assert_with_error!(&rewards_a, &expected_rewards_a, &tolerance_a, "User A proportional rewards");
+    assert_with_error!(&rewards_b, &expected_rewards_b, &tolerance_b, "User B proportional rewards");
+    
+}
+
+#[test]
+fn test_users_joining_different_times_fair_distribution() {
+    let (pic, canister_id, ledger_id) = setup();
+    let user_early = Principal::from_text("rdmx6-jaaaa-aaaaa-aaadq-cai").unwrap();
+    let user_late = Principal::from_text("rrkah-fqaaa-aaaaa-aaaaq-cai").unwrap();
+    
+    let deposit_amount = Nat::from(100_000_000u64);
+    let reward_amount = Nat::from(200_000_000u64);
+    
+    let stakable_episode = get_stakable_episode(&pic, canister_id, 7);
+    
+    create_deposit(&pic, canister_id, ledger_id, user_early, deposit_amount.clone(), stakable_episode);
+    
+    reward_pool(&pic, canister_id, ledger_id, user_early, reward_amount.clone())
+        .expect("Reward pool should succeed");
+    
+    let reward_time = pic.get_time().as_nanos_since_unix_epoch() / 1_000_000_000;
+    let last_reward_episode = (reward_time + EPISODE_DURATION * 12) / EPISODE_DURATION;
+    let exact_reward_duration = (last_reward_episode + 1) * EPISODE_DURATION - reward_time;
+    
+    advance_time(&pic, exact_reward_duration / 4);
+    
+    create_deposit(&pic, canister_id, ledger_id, user_late, deposit_amount.clone(), stakable_episode);
+    
+    advance_time(&pic, (exact_reward_duration * 3) / 4);
+    
+    let rewards_early = pic
+        .query_call(canister_id, user_early, "get_deposits_rewards", encode_args((vec![0u64],)).unwrap())
+        .map(|r| decode_one::<Nat>(&r).unwrap())
+        .expect("Failed to get early user rewards");
+    
+    let rewards_late = pic
+        .query_call(canister_id, user_late, "get_deposits_rewards", encode_args((vec![1u64],)).unwrap())
+        .map(|r| decode_one::<Nat>(&r).unwrap())
+        .expect("Failed to get late user rewards");
+    
+    let expected_early = (reward_amount.clone() * Nat::from(625u64)) / Nat::from(1000u64);
+    
+    let expected_late = (reward_amount.clone() * Nat::from(375u64)) / Nat::from(1000u64);
+    
+    assert_with_error!(&rewards_early, &expected_early, &ALLOWED_ERROR, "Early user fair rewards");
+    assert_with_error!(&rewards_late, &expected_late, &ALLOWED_ERROR, "Late user fair rewards");
+    
+}
+#[test]
+fn test_reward_withdrawal_ownership_and_security() {
+    let (pic, canister_id, ledger_id) = setup();
+    let user_a = Principal::from_text("rdmx6-jaaaa-aaaaa-aaadq-cai").unwrap();
+    let user_b = Principal::from_text("rrkah-fqaaa-aaaaa-aaaaq-cai").unwrap();
+    let malicious_user = Principal::from_text("ryjl3-tyaaa-aaaaa-aaaba-cai").unwrap();
+    
+    let deposit_amount = Nat::from(100_000_000u64);
```

Comment:
In all the test when we work with fake bitcoins show in comments what is the expected amount of it in btc
So, here just add comment : // 1 BTC
the same in other places in rewards test module.